### PR TITLE
BW-1297 Auto-publishing of CBAS-UI image to GCR

### PR DESCRIPTION
**Pull request by fredlas**
Commits to main will now trigger a (tag-incremented) new image to be pushed to dockerhub and GCR.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        - name: Build Docker image
+          run: docker build . --no-cache -t ${{ steps.image-name.outputs.DOCKERHUB_NAME }}
+
+        - name: Login to Docker Hub
+          uses: docker/login-action@v1
+          with:
+            username: dsdejenkins
+            password: ${{ secrets.DSDEJENKINS_DOCKERHUB_PASSWORD }}
+
+        - name: Push dockerhub image
```
Comment (fredlas) [inline .github/workflows/publish.yml:60]: The instructions in this repo's readme, which describe building an image and pushing it to GCR, don't mention dockerhub. Should I take these dockerhub steps out?

Comment (fredlas) [reply .github/workflows/publish.yml:60]: (the answer was yes)

File: README.md [hunk 2]
```diff
@@ -57,6 +57,8 @@ Web user interface for Batch Analysis
 
 ### Build and publish Docker image
 
+(You should not need to manually build or publish the image: it should happen automatically after your PR is merged to main).
+
 From the root of the repository run the below steps:
 1. To build the docker image run: `docker build . --no-cache -t us.gcr.io/broad-dsp-gcr-public/terra-batch-analysis-ui:<tag>`
```
Comment (salonishah11) [inline README.md:63]: While you are changing this file, can you also change `terra-batch-analysis-ui` to `cbas-ui` here and below?

File: README.md [hunk 3]
```diff
@@ -57,6 +57,8 @@ Web user interface for Batch Analysis
 
 ### Build and publish Docker image
 
+(You should not need to manually build or publish the image: it should happen automatically after your PR is merged to main).
```
Comment (aednichols) [inline README.md:60]: Not sure this is completely true, sometimes one wants to build an image from not yet merged code for testing.

**Review by aednichols — APPROVED** [review]

File: .github/workflows/setup-and-test.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+
+    if: success() && github.ref == 'refs/heads/main'
+
+    steps:
+      - name: Fire off tag action
+        uses: broadinstitute/workflow-dispatch@v1
+        with:
+          workflow: Tag
+          token: ${{ secrets.BROADBOT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (fredlas) [reply .github/workflows/setup-and-test.yml:35]: Thanks, would have been lost without those links. Opened https://github.com/broadinstitute/terraform-ap-deployments/pull/729

[the remark this answers is not in the document]
Comment (fredlas) [reply .github/workflows/setup-and-test.yml:35]: Done.

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**